entropy/coin_flip/app/src/flip_coin.ts

```diff
@@ -29,75 +45,80 @@ const parser = yargs(hideBin(process.argv))
 
 async function main() {
   const argv = await parser.argv;
+  const chainName = argv.chainName;
+  if (chainName !== "optimism-sepolia" && chainName !== "arbitrum-sepolia") {
+    throw new Error("Invalid chain name");
+  }
+
+  let chain: Chain;
+  if (chainName === "optimism-sepolia") {
+    chain = optimismSepolia;
+  } else {
+    chain = arbitrumSepolia;
+  }
```

Comment:
More idiomatic would be to use a ternary:

```ts
const chain = chainName === "optimism-sepolia" ? optimismSepolia : arbitrumSepolia
```

But a still more idiomatic way to do this whole thing would be to use a record:

```ts
const main = async () => {
    const chain = CHAIN_NAME_TO_CHAIN[argv.chainName];
    if (chain === undefined) {
        throw new InvalidChainError(argv.chainName);
    }
    ...
}

const CHAIN_NAME_TO_CHAIN = {
    "optimism-sepolia": optimismSepolia,
    "arbitrum-sepolia": arbitrumSepolia
};

class InvalidChainError extends Error {
    override constructor(chainName: string) {
        super(`Invalid chain name: ${chainName}`);
    }
}
```

Reply:
Yes, that's the way. Previously I was using default rpc by viem, but they are slow. Because of that, I was assigning transport as well based on the chain.

Reply:
For that kind of thing you can still use a map, just map to a tuple:

```ts
const [chain, rpc] = CHAIN_NAME_TO_CHAIN_CONFIG[argv.chainName];
...
const CHAIN_NAME_TO_CHAIN_CONFIG = {
    "optimism-sepolia": [optimismSepolia, "https://some-rpc.com"]
} as const;
```